CMakePresets.json

```diff
@@ -4,17 +4,14 @@
   "configurePresets": [
     {
       "name": "default",
-      "displayName": "Automatic options (debug with tests)",
+      "displayName": "Automatic options",
       "description": "Dependencies are enabled based on environment probing",
       "binaryDir": "${sourceDir}/build-${presetName}",
       "generator": "Ninja",
       "cacheVariables": {
-        "BUILD_SHARED_LIBS":    {"type": "BOOL",   "value": "ON"},
         "Celeritas_GIT_DESCRIBE": "",
         "CELERITAS_BUILD_TESTS":{"type": "BOOL",   "value": "ON"},
-        "CELERITAS_DEBUG":      {"type": "BOOL",   "value": "ON"},
```

Comment:
No longer default to debug, because some people might want to use production by default. Now daughter configurations explicitly have to inherit from ndebug/reldeb/debug .